### DISCUSSION TITLE
win.PointerEvent is not a constructor

### DISCUSSION THREAD

**Discussion by christianhegedues** [chakra-ui/zag#1560] [General]
# 🐛 Bug report

Currently running tests that includes a Zagjs components throw the following error: `TypeError: win.PointerEvent is not a constructor`.  The error comes from the file `track-press.ts` of the @zag-js/dom-events and only occurs if you execute an event on another element, for example `document.body`.

## 💥 Steps to reproduce

1. Include, for example, @zag-js/checkbox in da test file
2. Fire an event on a different element, for example the `document.body`
3. Run the test file

## 💻 Link to reproduction

CodeSandbox reproduction: https://codesandbox.io/p/devbox/fervent-cdn-wg2lvd

## 🧐 Expected behavior

Tests should not throw an error when an event is executed on an element

## 🌍 System information

| Software         | Version(s) |
| ---------------- | ---------- |
| Zag Version      |     0.53.0       |

**Comment by segunadebayo** [comment]
This is a pending bug from jsdom, [see the issue](https://github.com/jsdom/jsdom/issues/2527)

Would recommend trying happy-dom instead, see if it works better.

**Reply by christianhegedues** [reply]
@segunadebayo with happy-dom the tests for the component seem to work properly. 
Many thanks for the clarification and the hint.
